lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -320,13 +320,27 @@ export class UserAgentApplication {
     }
 
     // #endregion
+
     /**
      * Use when initiating the login process by redirecting the user's browser to the authorization endpoint.
      * @param {@link (AuthenticationParameters:type)}
      */
     loginRedirect(userRequest?: AuthenticationParameters): void {
+        // Check if userRequest is empty
+        if (!userRequest) {
+            userRequest = {};
+        }
+        
+        // Check if request scopes are null
+        if (!userRequest.scopes) {
+            userRequest.scopes = [];
+        }
```

Comment:
Instead of creating this `userRequest`, I would prefer to see a new array created (and then have downstream code properly handle a falsey `userRequest`:

```js
const scopes = (userRequest && userRequest.scopes) || [];
```

Reply:
Would a ternary operator work better here? I assume this would set scopes to a boolean "true" when userRequest and userRequest.scopes are truthy, rather than set scopes to userRequest.scopes.

Reply:
Nvm, thanks for the explanation on this JS functionality.